src/main/java/com/epam/upskillproject/controller/command/ActionFactory.java

```
@@ -0,0 +1,238 @@
+        this.adminProfileCommand = adminProfileCommand;
+        this.adminUpdateCommand = adminUpdateCommand;
+        this.adminDeleteCommand = adminDeleteCommand;
+        this.customerListCommand = customerListCommand;
+        this.customerProfileCommand = customerProfileCommand;
+        this.customerUpdateCommand = customerUpdateCommand;
+        this.customerDeleteCommand = customerDeleteCommand;
+        this.customerAccountsCommand = customerAccountsCommand;
+        this.paymentListCommand = paymentListCommand;
+        this.systemIncomeCommand = systemIncomeCommand;
+        this.myAccountsCommand = myAccountsCommand;
+        this.myAccountsServiceCommand = myAccountsServiceCommand;
+        this.blockUserAccountCommand = blockUserAccountCommand;
+        this.deleteUserAccountCommand = deleteUserAccountCommand;
+        this.addUserAccountCommand = addUserAccountCommand;
+        this.blockUserCardCommand = blockUserCardCommand;
+        this.deleteUserCardCommand = deleteUserCardCommand;
+        this.addUserCardCommand = addUserCardCommand;
+        this.topupUserAccountCommand = topupUserAccountCommand;
+        this.performPaymentCommand = performPaymentCommand;
+        this.myAccountIncomingCommand = myAccountIncomingCommand;
+        this.myAccountOutgoingCommand = myAccountOutgoingCommand;
+        this.indexCommand = indexCommand;
+        this.logoutCommand = logoutCommand;
+        this.loginErrorCommand = loginErrorCommand;
+        this.signUpCommand = signUpCommand;
+        this.userProfileCommand = userProfileCommand;
+        this.userUpdateCommand = userUpdateCommand;
+        this.changeLangCommand = changeLangCommand;
+    }
+
+    public Command produce(HttpServletRequest req, HttpServletResponse resp) {
+        if (req != null && resp != null) {
+            try {
+                Optional<TargetType> target = paramReader.readTargetTypeParamOrAttr(req, TARGET_PARAM);
+                if (target.isPresent()) {
```

if not presented - return null. Should return an exception?